src/net_processing.cpp

```diff
@@ -5008,7 +5015,7 @@ void PeerManagerImpl::EvictExtraOutboundPeers(std::chrono::seconds now)
             if (pnode->GetId() > youngest_peer.first) {
                 next_youngest_peer = youngest_peer;
                 youngest_peer.first = pnode->GetId();
-                youngest_peer.second = pnode->m_last_block_time;
+                youngest_peer.second = *Assert(m_evictionman.GetLastBlockTime(pnode->GetId()));
```

Comment:
Repeatedly looking up the eviction manager map and locking and unlocking access to it, and introducing the potential for maps to be out of sync and having to add asserts to check for that doesn't seem ideal...